modules/apps/notification/notification-web/src/main/java/com/liferay/notification/web/internal/portlet/action/GetNotificationTemplateInitDataMVCResourceCommand.java

```diff
@@ -0,0 +1,185 @@
+ * details.
+ */
+
+package com.liferay.notification.web.internal.portlet.action;
+
+import com.liferay.notification.constants.NotificationPortletKeys;
+import com.liferay.object.definition.notification.term.util.ObjectDefinitionNotificationTermUtil;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectField;
+import com.liferay.object.model.ObjectRelationship;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.object.service.ObjectRelationshipLocalService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.feature.flag.FeatureFlagManagerUtil;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.security.auth.PrincipalThreadLocal;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + NotificationPortletKeys.NOTIFICATION_TEMPLATES,
+		"mvc.command.name=/notification_templates/get_notification_template_init_data"
+	},
+	service = MVCResourceCommand.class
+)
+public class GetNotificationTemplateInitDataMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		_objectDefinition = _objectDefinitionLocalService.fetchObjectDefinition(
+			ParamUtil.getLong(resourceRequest, "objectDefinitionId"));
+
+		if (_objectDefinition == null) {
+			return;
+		}
+
+		User user = _userLocalService.getUser(PrincipalThreadLocal.getUserId());
+
+		Map<String, String> authorObjectFieldNames = HashMapBuilder.put(
+			"author-email-address", "AUTHOR_EMAIL_ADDRESS"
+		).put(
+			"author-first-name", "AUTHOR_FIRST_NAME"
+		).put(
+			"author-id", "AUTHOR_ID"
+		).put(
+			"author-last-name", "AUTHOR_LAST_NAME"
+		).put(
+			"author-middle-name", "AUTHOR_MIDDLE_NAME"
+		).put(
+			"author-prefix", "AUTHOR_PREFIX"
+		).put(
+			"author-suffix", "AUTHOR_SUFFIX"
+		).build();
```

Comment:
@p-albuquerque can we extend `BaseNotificationTemplateTermsMVCResourceCommand` to use the existing `authorObjectFieldNames` variable there?

Reply:
I didn't want to extend it in a first approach, because the kind of _javascript object_ the childs of this abstract class return have a structure a bit different, then, I will need to reshape this childs, and it means changes in **frontend** right now. 

I could also keep the already existing structures intact making an  _Override_ entirely the **doServeResource** in this new one child, but I will need to change the signature of method `getTermNamesEntries()` since it will not implement it.

I'm not sure what is the most indicated.